core/crypto/src/key_file.rs

```diff
@@ -16,18 +16,16 @@ pub struct KeyFile {
 }
 
 impl KeyFile {
-    pub fn write_to_file(&self, path: &Path) {
-        let mut file = File::create(path).expect("Failed to create / write a key file.");
+    pub fn write_to_file(&self, path: &Path) -> std::io::Result<()> {
+        let mut file = File::create(path)?;
         #[cfg(platform = "unix")]
         {
             use std::os::unix::fs::PermissionsExt;
             let perm = std::fs::Permissions::from_mode(u32::from(libc::S_IWUSR | libc::S_IRUSR));
-            file.set_permissions(perm).expect("Failed to set permissions for a key file.");
-        }
-        let str = serde_json::to_string_pretty(self).expect("Error serializing the key file.");
```

Comment:
It may be a good idea to print the path, where this operation failed. That, would make it easier to debug if something happens. Otherwise, you kind of have to modify the code, to figure out what failed.

Reply:
Ah yea so what I tried to do here was make the callers print out the path in config.rs, like:

```rust
network_signer
                .write_to_file(&path)
                .with_context(|| format!("Error writing key file to {}", path.display()))?
```
But now that I look at it again maybe it's easier to do that in this function here, yeah. Since we'd have to do that in every caller anyway. Maybe the way to go is to make this function return an `anyhow::Result<()>` and call `with_context` in here?